### PR TITLE
Tensor fusion for sharding stage 1

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR types
Others

### PR changes
Others

### Description
speed increase 1.5% with 8 acc steps.
image

### CODE REVIEW

File: paddlenlp/trainer/training_args.py [hunk 1]
```diff
@@ -820,6 +834,27 @@ def __post_init__(self):
 
                 # setter once https://github.com/PaddlePaddle/Paddle/blob/b7295120b0e78b293cd7ae29706e21769d06a3cc/python/paddle/distributed/fleet/base/distributed_strategy.py#L1692
                 strategy.hybrid_configs = hybrid_configs
+
+                if sharding_parallel_degree > 1:
+                    sharding_parallel_config = set(self.sharding_parallel_config.split(" "))
+                    for x in sharding_parallel_config:
+                        if len(x) > 0:
+                            if x not in [
+                                "sharding_stage_1_tensor_fusion",
```
Comment (ZHUI) [inline paddlenlp/trainer/training_args.py]: `sharding_stage_1_tensor_fusion`
-> `enable_stage1_tensor_fusion`

File: paddlenlp/trainer/training_args.py [hunk 2]
```diff
@@ -820,6 +834,27 @@ def __post_init__(self):
...
+                            ]:
+                                raise ValueError(
+                                    f"Found unknown pipeline mode config {x}, "
+                                    f"accpet config is sharding_stage_1_tensor_fusion."
+                                )
+                    try:
```
Comment (ZHUI) [inline paddlenlp/trainer/training_args.py:849]: 这里的话，要不限制一下stage1 吧。非stage1 开启，报个错？

Comment (FeixLiu) [reply paddlenlp/trainer/training_args.py:849]: 觉得不用报错？非stage 1配这个就是不生效

File: paddlenlp/trainer/training_args.py (hunk 1 above)
Comment (FeixLiu) [reply paddlenlp/trainer/training_args.py]: done update

**Review by ZHUI — APPROVED** [review]
LGTM

**Review by wawltor — APPROVED** [review]
LGTM

**Pull request merged**